providers/pipedrive/metadata_test.go

```diff
@@ -0,0 +1,160 @@
+package pipedrive
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/amp-labs/connectors"
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/test/utils/mockutils/mockserver"
+	"github.com/amp-labs/connectors/test/utils/testroutines"
+	"github.com/amp-labs/connectors/test/utils/testutils"
+)
+
+func TestListObjectMetadata(t *testing.T) { // nolint:funlen,gocognit,cyclop
+	t.Parallel()
+
+	zeroRecords := testutils.DataFromFile(t, "zero-records.json")
+	success := testutils.DataFromFile(t, "currencies.json")
+
+	tests := []testroutines.Metadata{
+		{
+			Name:         "At least one object name must be provided",
+			Input:        nil,
+			Server:       mockserver.Dummy(),
+			ExpectedErrs: []error{common.ErrMissingObjects},
+		},
+		{
+			Name:  "A success API Response",
+			Input: []string{"currencies"},
+			Server: httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.Header().Set("Content-Type", "application/json")
+				w.WriteHeader(http.StatusOK)
+				_, _ = w.Write([]byte(success))
+			})),
+			Expected: &common.ListObjectMetadataResult{
+				Result: map[string]common.ObjectMetadata{
+					"currencies": {
+						DisplayName: "currencies",
+						FieldsMap: map[string]string{
+							"active_flag":    "active_flag",
+							"code":           "code",
+							"decimal_points": "decimal_points",
+							"id":             "id",
+							"is_custom_flag": "is_custom_flag",
+							"name":           "name",
+							"symbol":         "symbol",
+						},
+					},
+				},
+				Errors: map[string]error{},
+			},
+			ExpectedErrs: nil,
+		},
+		{
+			Name:  "Zero records returned from server",
+			Input: []string{"activities"},
+			Server: httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.Header().Set("Content-Type", "application/json")
+				w.WriteHeader(http.StatusOK)
+				_, _ = w.Write([]byte(zeroRecords))
+			})),
+			Expected: &common.ListObjectMetadataResult{
```

Comment:
You could shrink the output by comparing only the subset of fields: 
```
Comparator: func(baseURL string, actual, expected *common.ListObjectMetadataResult) bool {
    return mockutils.MetadataResultComparator.SubsetFields(actual, expected)
},
```

Reply:
Cool, Looking into updating/cleaning these tests, As am working on adding Mock Tests for the other connectors.